### ISSUE TITLE
Reopening ActionMailer::DeliveryJob to add sidekiq retry logic fails

### ISSUE THREAD

**Issue by bdmac** [mperham/sidekiq#2740]
I'm not entirely sure where this problem lies but I attempted to simply reopen the ActionMailer::DeliveryJob to add in a `rescue_from ActiveJob::DeserializationError` so I could retry the mail delivery but that lead to an ArgumentError on the call to perform when sidekiq retried the delivery.

```
13:35:14 worker.1 | 2015-12-29T21:35:14.024Z 77615 TID-oup9gadwg WARN: {"class"=>"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper", "wrapped"=>"ActionMailer::DeliveryJob", "queue"=>"mailers", "args"=>[{"job_class"=>"ActionMailer::DeliveryJob", "job_id"=>"93f7416e-94ad-4cd1-bb74-96d89091c4f6", "queue_name"=>"mailers", "arguments"=>[], "locale"=>"en"}], "retry"=>true, "jid"=>"28da06689bff4e78c72b99db", "created_at"=>1451424880.1423612, "locale"=>"en", "enqueued_at"=>1451424911.560232, "error_message"=>"wrong number of arguments (0 for 3+)", "error_class"=>"ArgumentError", "failed_at"=>1451424914.02438, "retry_count"=>0}
13:35:14 worker.1 | 2015-12-29T21:35:14.024Z 77615 TID-oup9gadwg WARN: ArgumentError: wrong number of arguments (0 for 3+)
13:35:14 worker.1 | 2015-12-29T21:35:14.025Z 77615 TID-oup9gadwg WARN: /Users/bdmac/.gem/ruby/2.2.2/gems/actionmailer-4.2.4/lib/action_mailer/delivery_job.rb:9:in `perform'
```

As you can see Sidekiq's serialized job has empty arguments which I presume is why the retry call to perform fails since ActionMailer::DeliveryJob#perform expects 3 or more arguments.

The monkey-patch was pretty straightforward just reopening the class and adding in the `rescue_from` call/block to trigger `retry_job wait: 10.seconds`.

**Comment by mperham** [comment]
Could you clarify?  Why is this a Sidekiq issue?  Sidekiq has almost no ActiveJob-specific code.

**Comment by bdmac** [comment]
As I said I'm not entirely sure it is but I did notice the arguments are lost. Thought it might be something with the adapter between the two. Who maintains the adapter code for sidekiq<->activejob?

**Comment by mperham** [comment]
The adapter is shipped with activejob so that would be rails core. https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb

**Comment by janosch-x** [comment]
I searched quite a while to figure this one out. Others might find it helpful to know that it is fixed in rails 5.2.1

https://github.com/rails/rails/issues/22044
